jarviscli/plugins/scan_network.py

```diff
@@ -0,0 +1,51 @@
+import nmap
+import json
+from plugin import plugin, require, LINUX
+from colorama import Fore
+
+
+def error_output(jarvis, nm_scan):
+    jarvis.say('Please, check your input is correct\n', Fore.RED)
+    jarvis.say('This might hint what the problem is:\n', Fore.YELLOW)
+    jarvis.say(str(nm_scan.scaninfo()['error']), Fore.YELLOW)
+
+
+def show_output(jarvis, results):
+    uphosts = results['nmap']['scanstats']['uphosts']
+    jarvis.say('Number of hosts up: ' + uphosts + '\n', Fore.GREEN)
+    devices = results['scan']
+    for d in devices:
+        jarvis.say(d, Fore.GREEN)
+        jarvis.say('\tName: ' + devices[d]['hostnames'][0]['name'], Fore.GREEN)
+        tcp_ports = devices[d]['tcp']
+        for port in tcp_ports:
+            jarvis.say('\tOpen TCP Port: ' + str(port), Fore.GREEN)
+            jarvis.say('\t\tProtocol Name: ' + tcp_ports[port]['name'], Fore.GREEN)
+            jarvis.say('\t\tProduct: ' + tcp_ports[port]['product'], Fore.GREEN)
+            jarvis.say('\t\tVersion: ' + tcp_ports[port]['version'], Fore.GREEN)
+        print('----------------------------\n')
+
+
+@require(network=True)
+@require(native="nmap", platform=LINUX)
+@plugin('scan_network')
+def scan(jarvis, s):
+    """
+    Scans provided network for all connected devices
+    Usage example: jarvis scan_network 172.16.1.10/24
+    """
+    nm_scan = nmap.PortScanner()
+    jarvis.say('Please wait, this might take a while!')
+    if not s:
+        jarvis.say('You need to provide address as input!\n'
+                   ' For example: jarvis scan_network x.x.x.x/24', Fore.YELLOW)
+    else:
+        nm_scan.scan(s)
```

Comment:
Noticed python-nmap default adds '-sV' as option. According to man

``-sV: Probe open ports to determine service/version info``

This takes ages and in line 49 you are dropping this information anyway. As a suggestion, overwrite this parameter for extremely faster scan times:
```suggestion
        nm_scan.scan(s, arguments="")
```

Reply:
Thanks for pointing that out!

Reply:
So I checked this in detail and although removing the arguments provides a significant speedup, there is a miss on important information. While I do remove the services, version info provides helpful information about the underlying operating system. Unfortunately, I could find a way to extract the version only - without the services; thus I am going to leave the default arguments and just place a comment providing your suggestion for people who would favor speed over detail depth.